File: Server/src/plugin/interaction/city/falador/WineOfZamorakInteraction.java
package plugin.interaction.city.falador;

import org.crandor.game.interaction.Option;
import org.crandor.game.interaction.SpecialGroundInteraction;
import org.crandor.game.interaction.SpecialGroundItems;
import org.crandor.game.node.entity.npc.NPC;
import org.crandor.game.node.entity.player.Player;
import org.crandor.game.world.map.RegionManager;

import java.util.List;

public class WineOfZamorakInteraction extends SpecialGroundInteraction {
    @Override
    public void handle(Player player, Option option){
        player.faceLocation(SpecialGroundItems.WINE_OF_ZAMORAK.getLocation());
        final List<NPC> npcs = RegionManager.getLocalNpcs(player);
        for (NPC n : npcs) {
            if (n.getId() == 188) {
                n.sendChat("Hands off zamorak's wine!");
                n.getProperties().getCombatPulse().attack(player);
            }
        }
    }
}

File: Server/src/main/content/region/asgarnia/handlers/WineOfZamorakInteraction.kt
package content.region.asgarnia.handlers

import core.game.node.item.GroundItem
import core.game.world.map.RegionManager
import org.rs09.consts.Items
import core.game.global.action.PickupHandler
import core.game.interaction.InteractionListener
import core.game.interaction.IntType

class WineOfZamorakInteraction : InteractionListener {

    override fun defineListeners() {
        on(Items.WINE_OF_ZAMORAK_245, IntType.GROUNDITEM,"take"){ player, wine ->
            if(player.location.regionId != 11574){
                PickupHandler.take(player, wine as GroundItem)
                return@on true
            }
            val npcs = RegionManager.getLocalNpcs(player)
            for (n in npcs) {
                if (n.id == 188) {
                    n.sendChat("Hands off zamorak's wine!")
                    n.properties.combatPulse.attack(player)
                }
            }
            return@on true
        }
    }
}
